File: src/main/java/Api/model/Hour.java
package Api.model;

import java.sql.Date;
import java.sql.Time;

/**
 * Created by Boy, Wytze, Miguel.
 * This model represents the hour
 */
public class Hour {


    private String hour_approved;
    private int hour_subproject_number;
    private int hour_employee_number;
    private Time startTime;
    private Time endTime;
    private Time hour_amount_of_hours;
    private String hour_comments;
    private Date hour_date;
    private int id;

    public Hour(String hour_approved, int hour_subproject_number, int hour_employee_number, Time startTime, Time endTime, Time hour_amount_of_hours, String hour_comments, Date hour_date, int id) {
        this.hour_approved = hour_approved;
        this.hour_subproject_number = hour_subproject_number;
        this.hour_employee_number = hour_employee_number;
        this.startTime = startTime;
        this.endTime = endTime;
        this.hour_amount_of_hours = hour_amount_of_hours;
        this.hour_comments = hour_comments;
        this.hour_date = hour_date;
        this.id = id;
    }

    public Hour(int hour_subproject_number, int hour_employee_number, Time startTime, Time endTime, String hour_comments, Date hour_date) {
        this.hour_subproject_number = hour_subproject_number;
        this.hour_employee_number = hour_employee_number;
        this.startTime = startTime;
        this.endTime = endTime;
        this.hour_comments = hour_comments;
        this.hour_approved = "pending";
        hour_amount_of_hours = new Time(0);
        this.hour_amount_of_hours.setTime(endTime.getTime() - startTime.getTime() - 3600000);
        this.hour_date = hour_date;

    }

    public String hour_approved(){ return hour_approved;}

    public int getHour_subproject_number() {
        return hour_subproject_number;
    }

    public int getHour_employee_number() {
        return hour_employee_number;
    }

    public Time getStartTime() {
        return startTime;
    }

    public Time getEndTime() {
        return endTime;
    }

    public Time getHour_amount_of_hours() {
        return hour_amount_of_hours;
    }

    public String getHour_comments() {
        return hour_comments;
    }

    public Date getHour_date() {
        return hour_date;
    }
    public String getHour_approved() {
        return hour_approved;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setHour_approved(String hour_approved) {
        this.hour_approved = hour_approved;
    }

}

File: Client/src/app/models/Hour.ts
export class Hour{
  constructor(private _hour_approved:string,
  private _hour_subproject_number:number,
  private _hour_employee_number:number,
  private _startTime:string,
  private _endTime:string,
  private _hour_amount_of_hours:string,
  private _hour_comments:string,
  private _hour_date:string,
  private _id:number){

  }

  get hour_approved(): string {
    return this._hour_approved;
  }

  set hour_approved(value: string) {
    this._hour_approved = value;
  }

  get hour_subproject_number(): number {
    return this._hour_subproject_number;
  }

  set hour_subproject_number(value: number) {
    this._hour_subproject_number = value;
  }

  get hour_employee_number(): number {
    return this._hour_employee_number;
  }

  set hour_employee_number(value: number) {
    this._hour_employee_number = value;
  }

  get startTime(): string {
    return this._startTime;
  }

  set startTime(value: string) {
    this._startTime = value;
  }

  get endTime(): string {
    return this._endTime;
  }

  set endTime(value: string) {
    this._endTime = value;
  }

  get hour_amount_of_hours(): string {
    return this._hour_amount_of_hours;
  }

  set hour_amount_of_hours(value: string) {
    this._hour_amount_of_hours = value;
  }

  get hour_comments(): string {
    return this._hour_comments;
  }

  set hour_comments(value: string) {
    this._hour_comments = value;
  }

  get hour_date(): string {
    return this._hour_date;
  }

  set hour_date(value: string) {
    this._hour_date = value;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }
}
